File: 02-campeonato/main.py
participantes = ["AAA", "BBB", "CCC", "DDD", "EEE", "FFF"]

def sorteo(participantes):
    if( len(participantes) %2 != 0 ):
        participantes.append("descansa")

    fixture=[]
    rivales = len(equipos)-1
    partidos = len(equipos) / 2

    for r in range(rivales):
        cantidad = rivales
        
		for p in range(partidos):
			fixture.extend([participantes[p]])
            fixture.extend([participantes[cantidad]]])
            cantidad -= 1

        participantes.insert(1, participantes[-1])	
        participantes.pop()
        fixture.extend(["##### jornada", r])

    return fixture

print(sorteo(participantes))

File: 02-campeonato/main.js
const sorteo = (equipos) => {

    if (equipos.length % 2 > 0) {
        equipos.push('Descansa');
    }

    let fixture = []
    let rivales = equipos.length - 1;
    let partidos = equipos.length / 2;


    for (let i = 0; i < rivales; i++) {
        fixture[i] = []
        for (let j = 0; j < partidos; j++) {
            fixture[i].push([equipos[j], equipos[rivales - j]]);
        }
        equipos.splice(1, 0, equipos.pop())
    }

    return fixture;
}

let participantes = [
    "AAA",
    "BBB",
    "CCC",
    "DDD",
    "EEE",
    "FFF"
]

console.log(sorteo(participantes))